aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -654,6 +654,70 @@ Tensor& lgamma_out(Tensor& result, const Tensor& self) { return unary_op_impl_fl
 Tensor lgamma(const Tensor& self) { return unary_op_impl_float(self, lgamma_stub); }
 Tensor& lgamma_(Tensor& self) { return unary_op_impl_(self, at::lgamma_out); }
 
+static inline TensorIterator build_frexp_iter(const Tensor &self,
+                                              Tensor mantissa, Tensor exponent) {
+
+  TensorIteratorConfig config;
+
+  config.set_check_mem_overlap(true)
+  .add_output(mantissa)
+  .add_output(exponent);
+
+  // gpu_kernel_multiple_outputs also has issues of handling non contiguous tensors
+  // Reference: https://github.com/pytorch/pytorch/pull/51097
+  if (self.device().type() == DeviceType::CUDA) {
+    config.add_input(self.contiguous());
```

Comment:
During my local development, I found the `gpu_kernel_multiple_outputs` does not behave correctly for `non-contiguous` tensors.
Calls `self.contiguous` before adding input to `TensorIterator` so that corresponding TestUnaryUfuncs checks would pass for CUDA path.

Reply:
This was due to compiler bug, and was fixed in #51827 (which also adds a test exercising non-contiguous tensors with gpu_kernel_multiple_outputs).

Reply:
@ngimel Thank you so much for this information.
I rebased my local development branch with latest code, the test cases related to `non-contiguous` tensors pass without calling `self.contiguous()` manullay.